lib/cli/daemoncommand.cpp

```diff
@@ -437,7 +437,17 @@ static pid_t StartUnixWorker(const std::vector<std::string>& configs, bool close
 		case -1:
 			Log(LogCritical, "cli")
 				<< "fork() failed with error code " << errno << ", \"" << Utility::FormatErrorNumber(errno) << "\"";
-			exit(EXIT_FAILURE);
+
+			try {
+				Application::InitializeBase();
+			} catch (const std::exception& ex) {
+				Log(LogCritical, "cli")
+					<< "Failed to re-initialize thread pool after forking (parent): " << DiagnosticInformation(ex);
+				exit(EXIT_FAILURE);
+			}
+
+			(void)sigprocmask(SIG_UNBLOCK, &l_UnixWorkerSignals, nullptr);
```

Comment:
This is now duplicated and could be moved outside of the `switch` statement to avoid this.

Reply:
How to not mix up the steps per case?

Reply:
I'm still in the process of figuring out how all this works. My idea would be to do this directly after the fork if we are in the parent. This would change when `Application::InitializeBase()` is executed though. From my current understanding, this just restarts some threads that were stopped before the fork, so I hope this should also be fine to restart these before the child signals us.

Reply:
As that code is also duplicated in the child case anyways, probably another issue.